### PR TITLE
Build sites dynamically with grunt

### PR DESCRIPTION
**Pull request by lukefrake**
### Blog title
Build sites dynamically with grunt
### Blog description
How to post on building multiple sites with Grunt.

#### Author
- [X] I have checked for grammatical errors
- [X] I have checked for spelling errors
- [X] All information (technical or otherwise) is correct to my knowledge

#### Editor 1
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

#### Editor 2
- [x] I have checked for grammatical errors
- [x] I have checked for spelling errors
- [x] All information (technical or otherwise) is correct to my knowledge

### CODE REVIEW

**Comment by stuartstobie** [comment]
Proofreading to be done, sweet, I'll add this to my to do list for the morning.

**Comment by hxpaul** [comment]
I'll take a review of this

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+layout: post
+published: true
+title: "Building sites dynamically with Grunt"
+author: Luke Frake
+categories: ""
+tags: "grunt gruntjs javascript"
+---
+
+
+I’m a Front End Developer at a company that sells theme park breaks. Our latest project, was a huge rewrite of our back end systems. My role in the project was mainly updating our front end, including removing our old [Smarty](http://www.smarty.net/) templates, and updating them to use a mixture of [Handlebars](handlebarsjs.com) via [Assemble](assemble.io) and [Dust.js](http://linkedin.github.io/dustjs/).
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: assemble.io and handlebars.js need http://
Also that second sentence feels too long to me.

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+As we serve up 6 sites that sell very similar products, we made the decision to use a single template group. We manage to share about ~95% of the CSS between the 6 sites, with ~5% being brand specific styles. We call each of these sites a “blueprint”.
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: do you want `about` and `~` there, when they are intended to mean the same thing.

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+We didn’t have much of a build process at the time, we were using [Grunt](http://gruntjs.com/), but only to compile and minify our Sass. So the first thing I started with was trying to make our build process awesome.
+
+We didn’t have much of a build process at the time, we were using Grunt, but only to compile and minify our Sass. So the first thing I started with was trying to make our build process awesome.
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: repeated this whole line

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+&#x27;dist/brandBar/css/main.min.css&#x27; : &#x27;src/styles/brandBar/default.scss&#x27;,
+&#x27;dist/brandBaz/css/main.min.css&#x27; : &#x27;src/styles/brandBaz/default.scss&#x27; 
+      }
+    }
+  }
+}</code></pre>
+<p>See the Pen <a href='http://codepen.io/lukefrake/pen/VYdJRb/'>VYdJRb</a> by Luke Frake (<a href='http://codepen.io/lukefrake'>@lukefrake</a>) on <a href='http://codepen.io'>CodePen</a>.</p>
+</div><script async src="//assets.codepen.io/assets/embed/ei.js"></script>
+
+I was watching my beautiful code become more and more disgusting by the minute, with every brand or new task I added taking me about half an hour or more, and explaining the process to other people was becoming more and more convoluted.
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: BUZZ repetition of "more and more"
image
Oh, no-one else listen to Just a Minute?
Replacing the second occurrence with "increasingly" or something might read better, not important though.

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+    },
+    dist: {
+      files: {
+&#x27;dist/brandFoo/css/main.min.css&#x27; : &#x27;src/styles/brandFoo/default.scss&#x27;,
+&#x27;dist/brandBar/css/main.min.css&#x27; : &#x27;src/styles/brandBar/default.scss&#x27;,
+&#x27;dist/brandBaz/css/main.min.css&#x27; : &#x27;src/styles/brandBaz/default.scss&#x27; 
+      }
+    }
+  }
+}</code></pre>
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: sometimes you put the trailing `;` in the code examples and sometimes not best to keep consistent imo

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+<div data-height="85" data-theme-id="12057" data-slug-hash="gbKVBq" data-default-tab="js" data-user="lukefrake" class='codepen'><pre><code>grunt.initConfig( {} );</code></pre>
+<p>See the Pen <a href='http://codepen.io/lukefrake/pen/gbKVBq/'>gbKVBq</a> by Luke Frake (<a href='http://codepen.io/lukefrake'>@lukefrake</a>) on <a href='http://codepen.io'>CodePen</a>.</p>
+</div><script async src="//assets.codepen.io/assets/embed/ei.js"></script>
+
+This means we can dynamically build the object, so thats what I did. I’ve changed my Gruntfile now, so at the top I create a brand array, and I also create a few empty objects.
+
+<div data-height="150" data-theme-id="12057" data-slug-hash="pvKMQg" data-default-tab="js" data-user="lukefrake" class='codepen'><pre><code>// Going to loop these later
+var brands = [&#x27;brandFoo&#x27;, &#x27;brandBar&#x27;, &#x27;brandBaz&#x27;];
+// Going to fill these later
+var sass = {}</code></pre>
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: missing trailing `;`

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+<p>See the Pen <a href='http://codepen.io/lukefrake/pen/pvKMQg/'>pvKMQg</a> by Luke Frake (<a href='http://codepen.io/lukefrake'>@lukefrake</a>) on <a href='http://codepen.io'>CodePen</a>.</p>
+</div><script async src="//assets.codepen.io/assets/embed/ei.js"></script>
+
+I then loop over the brands array, using Lo-Dash, to create my brand information.
+
+<div data-height="130" data-theme-id="12057" data-slug-hash="raKXQy" data-default-tab="js" data-user="lukefrake" class='codepen'><pre><code>_.forEach( brands, function( brand ) {
+  // Do some brand stuff
+}</code></pre>
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: missing trailing `);`
also wouldn't just `brands.forEach` look better here? No need to introduce another library for this example

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+<p>See the Pen <a href='http://codepen.io/lukefrake/pen/raKXQy/'>raKXQy</a> by Luke Frake (<a href='http://codepen.io/lukefrake'>@lukefrake</a>) on <a href='http://codepen.io'>CodePen</a>.</p>
+</div><script async src="//assets.codepen.io/assets/embed/ei.js"></script>
+
+Inside the loop, I can start pushing what I would normally write directly into my initConfig object.
+
+<div data-height="110" data-theme-id="12057" data-slug-hash="LErwXd" data-default-tab="js" data-user="lukefrake" class='codepen'><pre><code>sass[&#x27;dist/&#x27; + brand + &#x27;/css/main.min.css] = &#x27;src/styles/&#x27; + brand + &#x27;/default.scss&#x27;;</code></pre>
+<p>See the Pen <a href='http://codepen.io/lukefrake/pen/LErwXd/'>LErwXd</a> by Luke Frake (<a href='http://codepen.io/lukefrake'>@lukefrake</a>) on <a href='http://codepen.io'>CodePen</a>.</p>
+</div><script async src="//assets.codepen.io/assets/embed/ei.js"></script>
+
+This means that when I come to create my configObject, I can just do something like this.
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: seems a bit wordy, I think "I can do this" looks and reads better.

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+<div data-height="265" data-theme-id="12057" data-slug-hash="OPEKrY" data-default-tab="js" data-user="lukefrake" class='codepen'><pre><code>var configObject = {
+  sass: {
+    options: {
+      sourcemap: &#x27;none&#x27;
+    },
+    dist: {
+      files: sass
+    }
+  }
+}</code></pre>
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: missing trailing `;`

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+<p>See the Pen <a href='http://codepen.io/lukefrake/pen/OPEKrY/'>OPEKrY</a> by Luke Frake (<a href='http://codepen.io/lukefrake'>@lukefrake</a>) on <a href='http://codepen.io'>CodePen</a>.</p>
+</div><script async src="//assets.codepen.io/assets/embed/ei.js"></script>
+
+*Note that anything non brand specific goes into the configObject. e.g. you only want 1 lot of options.*
+
+If you log your configObject out, it would look more like this.
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: clearer to say console.log or inspect imo, just a suggestion not a blocker though

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+&#x27;dist/brandBar/css/main.min.css&#x27; : &#x27;src/styles/brandBar/default.scss&#x27;,
+&#x27;dist/brandBaz/css/main.min.css&#x27; : &#x27;src/styles/brandBaz/default.scss&#x27; 
+      }
+    }
+  }
+}</code></pre>
+<p>See the Pen <a href='http://codepen.io/lukefrake/pen/RNJXEq/'>RNJXEq</a> by Luke Frake (<a href='http://codepen.io/lukefrake'>@lukefrake</a>) on <a href='http://codepen.io'>CodePen</a>.</p>
+</div><script async src="//assets.codepen.io/assets/embed/ei.js"></script>
+
+The eagle eyed amongst you will notice that’s exactly what we had above. This means, that when I go to add a new brand, I only need to add it to the brands array, Grunt will do all the hard work for me.
```
Comment (hxpaul) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: eagle-eyed I think

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md (hunk 7 above)
Comment (lukefrake) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: It's because at the bottom I link the my working example, which I don't want to be too different from this.

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md (hunk 1 above)
Comment (lukefrake) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: I don't think it reads too long when you don't read it as markdown, personally.

**Comment by hxpaul** [comment]
so only the missing `http://` and the duplicate line are proper errors, the rest is optional

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md (hunk 8 above)
Comment (lukefrake) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: Disagree, I don't like anything to read too prescribed.

**Comment by lukefrake** [comment]
Thanks @hxpaul - I've amended/ replied on everything you said.

Some really good feedback thanks.

File: _posts/2015-03-02-build-sites-dynamically-with-grunt.md [hunk 12]
```diff
@@ -0,0 +1,114 @@
...
+
+<div data-height="130" data-theme-id="12057" data-slug-hash="raKXQy" data-default-tab="js" data-user="lukefrake" class='codepen'><pre><code>_.forEach( brands, function( brand ) {
+  // Do some brand stuff
+} );</code></pre>
+<p>See the Pen <a href='http://codepen.io/lukefrake/pen/raKXQy/'>raKXQy</a> by Luke Frake (<a href='http://codepen.io/lukefrake'>@lukefrake</a>) on <a href='http://codepen.io'>CodePen</a>.</p>
+</div><script async src="//assets.codepen.io/assets/embed/ei.js"></script>
+
+Inside the loop, I can start pushing what I would normally write directly into my initConfig object.
+
+<div data-height="110" data-theme-id="12057" data-slug-hash="LErwXd" data-default-tab="js" data-user="lukefrake" class='codepen'><pre><code>sass[&#x27;dist/&#x27; + brand + &#x27;/css/main.min.css] = &#x27;src/styles/&#x27; + brand + &#x27;/default.scss&#x27;;</code></pre>
```
Comment (HeinBehrenshx) [inline _posts/2015-03-02-build-sites-dynamically-with-grunt.md]: Missing closing ' 
&#x27;/css/main.min.css]

**Comment by HeinBehrenshx** [comment]
I am reviewing now

**Comment by stuartstobie** [comment]
I've become busier than I had thought I would since this morning. If anyone else wants to look at this go ahead.

**Comment by lukefrake** [comment]
@HeinBehrenshx - I've pushed up that quote now, thanks for that!

**Comment by HeinBehrenshx** [comment]
+1

**Comment by HeinBehrenshx** [comment]
+1
